Model/Product/Sync/Consumer.php

```diff
@@ -0,0 +1,129 @@
+use Magento\Framework\Json\Helper\Data as JsonHelper;
+use Nosto\Tagging\Logger\Logger;
+use Nosto\Tagging\Model\Product\Index\IndexRepository;
+use Nosto\Tagging\Model\Service\Sync as NostoSyncService;
+use Nosto\Tagging\Helper\Scope as NostoScopeHelper;
+
+/**
+ * Class Consumer
+ */
+class Consumer
+{
+    /** @var Logger */
+    private $logger;
+
+    /** @var JsonHelper */
+    private $jsonHelper;
+
+    /** @var IndexRepository */
+    private $indexRepository;
+
+    /** @var NostoSyncService */
+    private $nostoSyncService;
+
+    /** @var NostoScopeHelper */
+    private $nostoScopeHelper;
+
+    /** @var EntityManager */
+    private $entityManager;
+
+    /**
+     * Consumer constructor.
+     *
+     * @param Logger $logger
+     * @param JsonHelper $jsonHelper
+     * @param IndexRepository $indexRepository
+     * @param NostoSyncService $nostoSyncService
+     * @param NostoScopeHelper $nostoScopeHelper
+     * @param EntityManager $entityManager
+     */
+    public function __construct(
+        Logger $logger,
+        JsonHelper $jsonHelper,
+        IndexRepository $indexRepository,
+        NostoSyncService $nostoSyncService,
+        NostoScopeHelper $nostoScopeHelper,
+        EntityManager $entityManager
+    ) {
+        $this->logger = $logger;
+        $this->jsonHelper = $jsonHelper;
+        $this->indexRepository = $indexRepository;
+        $this->nostoSyncService = $nostoSyncService;
+        $this->nostoScopeHelper = $nostoScopeHelper;
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * Processing operation for product sync
+     *
+     * @param OperationInterface $operation
+     * @return void
+     * @throws Exception
+     */
+    public function processOperation(OperationInterface $operation)
+    {
+        $status = OperationInterface::STATUS_TYPE_COMPLETE;
+        $errorCode = null;
+        $message = null;
+        $serializedData = $operation->getSerializedData();
+        $unserializedData = $this->jsonHelper->jsonDecode($serializedData);
+        $productIds = $unserializedData['product_ids'];
+        $storeId = $unserializedData['store_id'];
+        try {
+            $store = $this->nostoScopeHelper->getStore($storeId);
+            $outOfSyncCollection = $this->indexRepository->getByProductIdsAndStoreId($productIds, $storeId);
+            $this->nostoSyncService->syncIndexedProducts($outOfSyncCollection, $store);
+            $this->nostoSyncService->syncDeletedProducts($store);
+        } catch (Exception $e) {
+            $this->logger->critical($e->getMessage());
+            $message = __('Something went wrong when syncing products to Nosto. Check log for details.');
+            $operation->setStatus(OperationInterface::STATUS_TYPE_NOT_RETRIABLY_FAILED)
+                ->setErrorCode($e->getCode())
+                ->setResultMessage($message);
+        }
+        $operation->setStatus($status ? OperationInterface::STATUS_TYPE_COMPLETE : null)
```

Comment:
The ternary part is redundant. You are defining the status already on line 106, it cannot be null / false.